app/assets/javascripts/discourse/app/components/user-preferences/user-passkeys.gjs

```diff
@@ -0,0 +1,252 @@
+  title = I18n.t("user.passkeys.title");
+  formatDate = getOwner(this).resolveRegistration("helper:format-date");
+  addedPrefix = I18n.t("user.passkeys.added_prefix");
+  lastUsedPrefix = I18n.t("user.passkeys.last_used_prefix");
+  neverUsed = I18n.t("user.passkeys.never_used");
+
+  isCurrentUser() {
+    return this.currentUser.id === this.args.model.id;
+  }
+
+  passkeyDefaultName() {
+    if (this.capabilities.isSafari) {
+      return I18n.t("user.passkeys.name.icloud_keychain");
+    }
+
+    if (this.capabilities.isAndroid || this.capabilities.isChrome) {
+      return I18n.t("user.passkeys.name.google_password_manager");
+    }
+
+    return I18n.t("user.passkeys.name.default");
+  }
+
+  async createPasskey() {
+    try {
+      const response = await this.args.model.createPasskey();
+
+      const publicKeyCredentialCreationOptions = {
+        challenge: Uint8Array.from(response.challenge, (c) => c.charCodeAt(0)),
+        rp: {
+          name: response.rp_name,
+          id: response.rp_id,
+        },
+        user: {
+          id: Uint8Array.from(response.user_secure_id, (c) => c.charCodeAt(0)),
+          name: this.currentUser.username,
+          displayName: this.currentUser.username,
+        },
+        pubKeyCredParams: response.supported_algorithms.map((alg) => {
+          return { type: "public-key", alg };
+        }),
+        excludeCredentials: response.existing_passkey_credential_ids.map(
+          (credentialId) => {
+            return {
+              type: "public-key",
+              id: stringToBuffer(atob(credentialId)),
+            };
+          }
+        ),
+        authenticatorSelection: {
+          // https://www.w3.org/TR/webauthn-2/#user-verification
+          // for passkeys (first factor), user verification should be marked as required
+          // it ensures browser prompts user for PIN/fingerprint/faceID before authenticating
+          userVerification: "required",
+          // See https://w3c.github.io/webauthn/#sctn-createCredential for context
+          // This ensures that the authenticator stores a client-side private key
+          // physical security keys (like Yubikey) need this
+          requireResidentKey: true,
+        },
+      };
+
+      const credential = await navigator.credentials.create({
+        publicKey: publicKeyCredentialCreationOptions,
+      });
+
+      let credentialParam = {
+        id: credential.id,
+        rawId: bufferToBase64(credential.rawId),
+        type: credential.type,
+        attestation: bufferToBase64(credential.response.attestationObject),
+        clientData: bufferToBase64(credential.response.clientDataJSON),
+        name: this.passkeyDefaultName(),
+      };
+
+      const registrationResponse = await this.args.model.registerPasskey(
+        credentialParam
+      );
+
+      if (registrationResponse.error) {
+        this.dialog.alert(registrationResponse.error);
+        return;
+      }
+
+      this.router.refresh();
```

Comment:
this returns a transition, I think it would be more correct to await here

Reply:
So, here the user is adding a new key, and we're about to show them a modal to name that key, but in the meantime, the key's already been stored. The call to `this.router.refresh` is updating the list of keys that is shown behind the dialog's overlay. In a sense, this is a background action, and the next item in the flow (showing the rename dialog) doesn't depend on the router refresh. In that sense, adding `await` isn't useful. (I do get that adding `await` would be more correct/consistent, but it would also be a tiny bit slower.)